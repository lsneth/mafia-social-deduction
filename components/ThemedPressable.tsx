import { forwardRef } from 'react'
import { Pressable, type PressableProps } from 'react-native'

export const ThemedPressable = forwardRef(function ThemedPressable(props: PressableProps, ref) {
  const { className, ...rest } = props
  return <Pressable className={`bg-red-800 p-3 w-full max-w-sm rounded-full ${className}`} {...rest} />
})
