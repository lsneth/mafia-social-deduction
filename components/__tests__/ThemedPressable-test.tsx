import * as React from 'react'
import renderer from 'react-test-renderer'

import { ThemedPressable } from '../ThemedPressable'

it(`renders correctly`, () => {
  const tree = renderer
    .create(<ThemedPressable>This is a unit test for the ThemedPressable component</ThemedPressable>)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
