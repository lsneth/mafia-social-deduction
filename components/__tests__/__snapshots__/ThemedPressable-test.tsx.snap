// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#991b1b",
      },
      {
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 6,
      },
      {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      },
      {
        "borderBottomLeftRadius": 9999,
        "borderBottomRightRadius": 9999,
        "borderTopLeftRadius": 9999,
        "borderTopRightRadius": 9999,
      },
    ]
  }
>
  This is a unit test for the ThemedPressable component
</View>
`;
